### DISCUSSION TITLE
Support SecretManagement

### DISCUSSION THREAD

**Discussion by akwebb1** [pester/vscode-adapter#82] [Q&A]
I tried the new adapter and it looks nice except for the fact that my tests use data stored in Microsoft.PowerShell.SecretManagement.  I can't find a way to incorporate that.   I just get an error that a password is required but I have no way of providing it that I have found.  Any suggestions?

**Comment by JustinGrote** [comment]
@akwebb1 try debugging a test, it will use the PSIC rather than the background and you will get a prompt.

Tests are not meant to be interactive, they're meant to run noninteractively, so you may want to reconsider your test by either mocking the secret, or using a value like the CredMan vault that doesn't require user interaction to retrieve a secret.

**Comment by akwebb1** [comment]
I will check out CredMan.  I usually just unlock my vault once in the VSCode PowerShell Integrated Console and then I can run/debug my tests without entering the password each time.

**Reply by JustinGrote** [reply]
The "run tests" Powershell engine uses a separate powershell runspace for performance, I do plan to add a setting to run the tests in the PSIC without debug eventually.

**Comment by akwebb1** [comment]
Cool.  I am trying to find info on the "CredMan vault" you mentioned which might be my solution but I am coming up empty.  Do you have a link?

**Comment by akwebb1** [comment]
is this it?  https://www.powershellgallery.com/packages/SecretManagement.JustinGrote.CredMan/1.0.0.  If so I am out of luck since I am running under both Windows and Linux.

**Reply by JustinGrote** [reply]
It is. You can look into the Azure Key Vault/LastPass/1Password/Keepass providers for a cross platform options that support authenticating with a cert or a stored credential.

**Comment by akwebb1** [comment]
I'll look into those as well.  Thanks for the help.
